### DISCUSSION TITLE
Serverless function is working locally but is not even showing up in the Vercel log in production

### DISCUSSION THREAD

**Discussion by Sounav-unstudio** [vercel/community#4355] [Help] [answered]
### Summary

I have a file called `cropImage.js` in `pages/api` folder. When I run it on my local server, it works fine as expected but recently it has not been sending a response back in my production environment, nor is it showing up on Vercel logs at all. After repeated calls, it is working once out of 5 times. I do not understand why. 

### Steps to Reproduce

Here's the code for the same
`import Jimp from 'jimp';
export const config = {
	api: {
		bodyParser: {
			sizeLimit: '35mb', // Set desired value here
		},
    responseLimit:'5mb'
	},
};

export default async function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).end();
  }
 <REDACTED INFORMATION>
  image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
    const alpha = this.bitmap.data[idx + 3];
    if (alpha > 0) {
      top = Math.min(top || y, y);
      left = Math.min(left || x, x);
      bottom = Math.max(bottom || y, y);
      right = Math.max(right || x, x);
    }
  });

  top = Math.max(top - margin, 0);
  left = Math.max(left - margin, 0);
  bottom = Math.min(bottom + margin, image.bitmap.height);
  right = Math.min(right + margin, image.bitmap.width);
  image.crop(left, top, right - left, bottom - top);
  console.log('Image cropped!')

  const croppedBuffer = await image.getBufferAsync(Jimp.MIME_PNG);
  console.log('Sending cropped image')

  return res.status(200).json({ image: croppedBuffer.toString('base64') });

}`

I have updated the vercel.json file to increase this particular function's max duration and memory to 45 seconds and 3008 MB respectively. But still, it does not work. It's not even showing up on the Vercel logs so I am not able to debug it as well

**Comment by nikoferro** [comment]
Happening the same here. Requests started timing out a couple of hours ago. When looking at the dev tools, it seems that the status code is sent back by the server but the content download stalls. Requests are not showing up on logs either

**Comment by amyegan** [comment]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to help with debugging?

**Comment by neelS-hah** [comment]
Bump. Same issue on our end too. The vercel deployment isn't showing any logs or errors.

**Comment by IsmaelP19** [comment]
I have the same issue. The endpoints of the backend that returns large amount of data (in my case it happens with large objects) are not returned and because of that, the serverless functions time out.
Vercel deployment doesn't show any specific log or error, just a 504 error on the request.

On local it works just fine, as it did on deployment a couple days ago.

**Reply by amyegan** [reply]
If the status is [504: FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#504:-function_invocation_timeout), that means the request reached the [timeout threshold](https://vercel.com/docs/limits/overview#serverless-function-execution-timeout). I don't have enough information about your site to determine the exact reason, but these are some possible causes:
> - The function is taking too long to process a request: Check that any API, or database requests you make in your function, are responding within the [Serverless Function Execution Duration](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) limit applicable to your plan.
> - Users on Pro and Enterprise plans can configure the maximum duration by using the maxDuration property. See the [maxDuration](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) section for more information
> - If you require a longer execution, [Edge Functions](https://vercel.com/docs/functions/edge-functions) provide an alternative that only require an [initial response](https://vercel.com/docs/functions/edge-functions/limitations#maximum-initial-response-time) within 25 seconds and allow you to [continue streaming responses](https://vercel.com/docs/functions/edge-functions/streaming) after that time.
> - The function isn't returning a response: The function must return an HTTP response, even if that response is an error. If no response is returned, the function will time out.
> - You have an infinite loop within your function: Check that your function is not making an infinite loop at any stage of execution.
> - Upstream errors: Check that any external API or database that you are attempting to call doesn't have any errors.
> - A common cause for this issue is when the application contains an unhandled exception. Check the application logs

https://vercel.com/guides/what-can-i-do-about-vercel-serverless-functions-timing-out

**Reply by IsmaelP19** [reply]
That's exactly what happens. My site is a social network for finding flatmates and when I try to get all the users (even though it is paginated to get only 5 users each time) for example, it does not return anything and so it times out. I thought this had to do with MongoDB Atlas, but on my local application, it connects just fine.
Besides, a couple days ago it was working as usual. This only happens when the response contains large amount of information (I don't know how much, but some next.js versions back, it told me that the response exceeded the threshold). When the response contains a tiny object it works fine.
I don't know how to debug the problem either since it works locally.

**Reply by amyegan** [reply]
Most applications work a little differently in local development mode than they do when run in a serverless environment. Your local computer doesn't have the same timeout limits for serverless functions, so that would explain why it doesn't run into a max duration limit on local but does in production. 

When accessing large amounts of data from a database or other source, retrieving the data from that external source could take long enough that it exceeds the Serverless Function [max duration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) limit before the data can be returned. This guide has some helpful info about how to fix function timeout errors: https://vercel.com/guides/what-can-i-do-about-vercel-serverless-functions-timing-out

You might also want to consider Edge Functions which don't have a set maximum duration. They must begin sending a response within 25 seconds and can continue [streaming](https://vercel.com/docs/functions/edge-functions/streaming) a response after that.

**Reply by IsmaelP19** [reply]
I know it's not the same on local than it is on deployment, but I got a little suspicious that it stopped working as it was a couple days ago.
Don't get me wrong, I appreciate your answers and I will keep looking for the problem. But changing all my application from Serverless Functions to Edge Functions is not a minor thing.

**Comment by IsmaelP19** [comment, marked as answer]
Working back again. Not fully functional but at least it's not showing up the 504 timeout error. Apparently, [this ](https://www.vercel-status.com/incidents/y1fn4l553q85)was the problem.
